pkg/pipeline/crd.go

```diff
@@ -35,39 +36,40 @@ import (
 const GenStatement = "// Code generated by terrajet. DO NOT EDIT."
 
 // NewCRDGenerator returns a new CRDGenerator.
-func NewCRDGenerator(groupDir, rootModulePath, group string) *CRDGenerator {
+func NewCRDGenerator(pkg *types.Package, localDirectoryPath, group string) *CRDGenerator {
 	return &CRDGenerator{
-		GroupDir:       groupDir,
-		RootModulePath: rootModulePath,
-		Group:          group,
+		LocalDirectoryPath: localDirectoryPath,
+		Group:              group,
+		pkg:                pkg,
 	}
 }
 
 // CRDGenerator takes certain information referencing Terraform resource definition
 // and writes kubebuilder CRD file.
 type CRDGenerator struct {
-	GroupDir       string
-	RootModulePath string
-	Group          string
+	LocalDirectoryPath string
+	Group              string
+
+	pkg *types.Package
 }
 
 // Generate builds and writes a new CRD out of Terraform resource definition.
 func (cg *CRDGenerator) Generate(version, kind string, schema *schema.Resource) error {
-	groupPkgPath := filepath.Join(cg.RootModulePath, "apis", strings.ToLower(strings.Split(cg.Group, ".")[0]), strings.ToLower(version))
-	kindPkgPath := filepath.Join(groupPkgPath, strings.ToLower(kind))
-	// TODO(muvaf): Loading the local package will result in error if there is
-	// any compilation errors, which is the case before running kubebuilder
-	// generators. For now, we act like the target package is empty.
-	pkg := types.NewPackage(kindPkgPath, strings.ToLower(version))
-	file := wrapper.NewFile(pkg.Path(), pkg.Name(), templates.CRDTypesTemplate,
+	file := wrapper.NewFile(cg.pkg.Path(), cg.pkg.Name(), templates.CRDTypesTemplate,
 		wrapper.WithGenStatement(GenStatement),
 		wrapper.WithHeaderPath("hack/boilerplate.go.txt"), // todo
-		wrapper.LinterEnabled(),
+		// wrapper.LinterEnabled(),
```

Comment:
Is this comment out intentional?

Reply:
there are other instances commented out as well in other files

Reply:
Just removed all. It turns out running `goimports` once for all files take much less time compared to having it run for every file before writing them to disk.